app/services/ingest_iati_activities.rb

```diff
@@ -13,15 +14,15 @@ def initialize(delivery_partner:, file_io:)
   def call
     doc = Nokogiri::XML(file_io, nil, "UTF-8")
 
-    legacy_activities = doc.xpath("//iati-activity")
+    legacy_activity_nodes = doc.xpath("//iati-activity")
+    legacy_activity_nodes.each do |legacy_activity_node|
+      legacy_activity = LegacyActivity.new(activity_node_set: legacy_activity_node)
 
-    legacy_activities.each do |legacy_activity|
```

Comment:
I thought that `legacy_activity` was too generic a name, I've tried to be more exact by including `node` in the names of these variables to communicate the type of variable.